### PR TITLE
[#131689197] RabbitMq Factory to use Promise V1.1

### PR DESCRIPTION
**Pull request by jonajonlee**
https://www.pivotaltracker.com/story/show/131689197

Fix bugs when trying to send a rabbitmq event
- rabbitmq/factory returns an object { send:Fn, listen:Fn, addQueue:Fn } where send and listen aren't handling getQueue as a promise returning function.
- GetQueue was expecting a name as string but got object

### CODE REVIEW

**Review by mattsmee — APPROVED** [review]
:shipit:

**Review by Kolossion — APPROVED** [review]
:shipit:

File: lib/memory/index.js [hunk 1]
```diff
@@ -123,6 +123,9 @@ module.exports = function MemoryStore(data_set) {
 
   });
 
+
+  const getState = R.always(store)
```
Comment (scull7) [inline lib/memory/index.js]: ahhh... it's everywhere!

File: lib/rabbitmq/factory.js [hunk 2]
```diff
@@ -22,14 +23,20 @@ const Factory = (config) => {
 
   return {
 
-    send: (name, msg, options = {}) =>
-      getQueue(name).send(options, msg)
+    send: (name, msg) =>
+      Bluebird.resolve(getQueue(name))
```
Comment (scull7) [inline lib/rabbitmq/factory.js]: what is going on here?

File: lib/memory/index.js (hunk 1 above)
Comment (jonajonlee) [reply lib/memory/index.js]: oh i thought this was merged a long time ago. I'll remove this

File: lib/rabbitmq/factory.js (hunk 2 above)
Comment (jonajonlee) [reply lib/rabbitmq/factory.js]: getQueue(name) was returning a promise so there wasn't a send property of it. I used Bluebird so I can have .get

Comment (scull7) [reply lib/rabbitmq/factory.js]: Isn't this a lot easier to understand?

``` javascript
send: (name, msg) => getQueue(name).then(q => q.send(name, msg))
```

Comment (jonajonlee) [reply lib/rabbitmq/factory.js]: true

Comment (jonajonlee) [reply lib/rabbitmq/factory.js]: updated

**Pull request merged**